imap/tls.c

```diff
@@ -1314,19 +1314,44 @@ EXPORTED int tls_start_servertls(int readfd, int writefd, int timeout,
     saslprops->ssf = (sasl_ssf_t) tls_cipher_usebits;
 
 #if (OPENSSL_VERSION_NUMBER >= 0x0090800fL)
-    if (SSL_session_reused(tls_conn)) {
-        saslprops->cbinding.len = SSL_get_finished(tls_conn,
-                                                   saslprops->tls_finished,
-                                                   MAX_FINISHED_LEN);
+    if (SSL_version(tls_conn) <= TLS1_2_VERSION) {
+        if (SSL_session_reused(tls_conn)) {
+            saslprops->cbinding.len = SSL_get_finished(tls_conn,
+                                                    saslprops->tls_finished,
+                                                    MAX_FINISHED_LEN);
+        }
+        else {
+            saslprops->cbinding.len = SSL_get_peer_finished(tls_conn,
+                                                    saslprops->tls_finished,
+                                                    MAX_FINISHED_LEN);
+        }
+
+        saslprops->cbinding.name = "tls-unique";
+        saslprops->cbinding.data = saslprops->tls_finished;
     }
+#if (OPENSSL_VERSION_NUMBER >= 0x10100001L)
```

Comment:
I think OpenSSL 3 can be assumed these days.  The OpenSSL 1.x series is end of life now.

Reply:
I agree. But as long as I can count over thirty entries with #if OPENSSL_VERSION_NUMBER ... in this repository, I will also add such compile switches. It's not my job to cleanup these compile switches and I cannot predict which company/user is still using older Linux distributions e.g. CentOS 7.4 with OpenSSL 1.0.2 or just using TLS 1.2.